src/aws-cpp-sdk-core/source/client/UserAgent.cpp

```diff
@@ -0,0 +1,166 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/core/Version.h>
+#include <aws/core/client/UserAgent.h>
+#include <aws/core/platform/Environment.h>
+#include <aws/core/platform/OSVersionInfo.h>
+#include <aws/core/utils/logging/LogMacros.h>
+#include <aws/core/utils/memory/stl/AWSStringStream.h>
+#include <aws/crt/Config.h>
+
+#include <numeric>
+#include <utility>
+
+using namespace Aws::Client;
+
+namespace {
+const char* LOG_TAG = "UserAgent";
+
+const std::pair<UserAgentFeature, const char*> BUSINESS_METRIC_MAPPING[] = {
+    {UserAgentFeature::RETRY_MODE_LEGACY, "D"}, {UserAgentFeature::RETRY_MODE_STANDARD, "E"}, {UserAgentFeature::RETRY_MODE_ADAPTIVE, "F"},
```

Comment:
please don't format like this, newline for each entry is okay
also, you may accidentally create an extra empty feature by using a TS-style extra last comma in the initializer list

Reply:
i actually had it that way before running clang format on the file, will do one perline (i like it), we can talk about the clang format update again soon